public/app/plugins/panel/heatmap/fields.ts

```diff
@@ -109,7 +112,7 @@ export function prepareHeatmapData(
   if (!rowsHeatmap) {
     if (frames.length > 1) {
       let allNamesNumeric = frames.every(
-        (frame) => !Number.isNaN(parseSampleValue(frame.name ?? frame.fields[1].name))
+        (frame) => !Number.isNaN(parseSampleValue(frame.fields[1].state?.displayName!))
```

Comment:
verified that removal of `frame.name` here does not regress the issues fixed by https://github.com/grafana/grafana/pull/65322